### PR TITLE
Fix for "Run Current Query" keybind no longer behaving as expected (#10538)

### LINKED ISSUE #10538: "Run Current Query" keybind no longer behaving as expected
Issue Type: <b>Bug</b>

In previous versions the expected behavior of this keybind was to run either the currently selected query or run whichever query the cursor was currently active on. This also used to be able to handle comments included within the queries and selections, as well as multiple queries in the same batch. There are a couple of easily reproducable issues now:

1) Pressing this keybind now runs the entire script regardless of cursor position.

2) Highlighting multiple queries and pressing this keybind results in only the top query being run. In the past, it would run the entire selection. Example: highlight and run both of these in the same batch
select 1
select 1

3) If there is a comment above a query and you include this in the highlighted text, the query will not run. Example: highlight the comment and the select and run.
-- the following gets 1
select 1

4) If you have a query commented out and highlight it to run, it will not run. Example:  run the select 1 by highlighting everything besides the comment characters
-- select 1

Azure Data Studio version: azuredatastudio 1.18.0 (2acd37a1b1cff34b64fbb4ea2d40715d7e2accac, 2020-05-19T23:54:30.554Z)
OS version: Windows_NT x64 10.0.18362

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-4570 CPU @ 3.20GHz (4 x 3193)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off_ok<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|11.91GB (5.14GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (17)</summary>

Extension|Author (truncated)|Version
---|---|---
bracket-pair-colorizer|Coe|1.0.60
gitlens|eam|9.8.1
ssmskeymap|kev|1.1.0
query-history|Mic|0.1.0
server-report|Mic|0.2.2
whoisactive|Mic|0.1.4
azure-account|ms-|0.8.4
powershell|ms-|2019.5.0
vsliveshare|ms-|1.0.337
team|ms-|1.149.2
azdata-sanddance|msr|1.3.9
sql-search|Red|0.3.3
simple-data-scripter|sea|0.1.0
vscodeintellicode|Vis|1.2.1
vscode-icons|vsc|8.8.0
snippet-creator|nik|0.0.4
poor-sql-formatter|WSR|0.1.0

(3 theme extensions excluded)

</details>

### PR DESCRIPTION
**Pull request by aasimkhan30**
…t statement in the selection

This PR fixes #10538

### CODE REVIEW

File: src/sql/workbench/contrib/query/browser/queryActions.ts [hunk 1]
```diff
@@ -224,7 +224,7 @@ export class RunQueryAction extends QueryTaskbarAction {
 		if (!this.editor.isSelectionEmpty()) {
 			if (this.isConnected(this.editor)) {
 				// If we are already connected, run the query
-				this.runQuery(this.editor, true);
+				this.runQuery(this.editor);
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/query/browser/queryActions.ts:227]: How is this fixing it? The parameter is literally called runCurrentStatement and defaults to false so this seems like the correct value to me. 

The bug reports seem to indicate that this only is happening when you don't have something selected. It's supposed to just run the single line you're on but it looks like that broke somewhere.

**Comment by aaomidi** [comment]
This doesn't seem to be a regression does it?

File: src/sql/workbench/contrib/query/browser/queryActions.ts (hunk 1 above)
Comment (aasimkhan30) [reply src/sql/workbench/contrib/query/browser/queryActions.ts:227]: Sorry about that. I confused this command with the run command. Making a new fix right now

**Comment by Charles-Gagnon** [comment]
Let's take this opportunity to add a test here for this

File: src/sql/workbench/contrib/query/test/browser/queryActions.test.ts [hunk 2]
```diff
@@ -560,6 +560,119 @@ suite('SQL QueryAction Tests', () => {
...
+
+		// Calling runCurrent with no open connection
+		let queryAction: RunQueryAction = new RunQueryAction(queryEditor.object, queryModelService.object, connectionManagementService.object);
+		calledRunQueryStatementOnInput = false;
+		await queryAction.runCurrent();
+
+		//connection dialog should open and runQueryStatement should not be called
+		assert.equal(calledRunQueryStatementOnInput, false, 'runCurrent should not call runQueryStatement');
+		testQueryInput.verify(x => x.runQueryStatement(TypeMoq.It.isAny()), TypeMoq.Times.never());
+		assert.equal(countCalledShowDialog, 1, 'showDialog is called');
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/query/test/browser/queryActions.test.ts:603]: The message should indicate what went wrong - in this case you expected it to be called but it wasn't.

File: src/sql/workbench/contrib/query/test/browser/queryActions.test.ts [hunk 3]
```diff
@@ -560,6 +560,119 @@ suite('SQL QueryAction Tests', () => {
...
+
+		// mocking query editor
+		const contextkeyservice = new MockContextKeyService();
+		let queryEditor = TypeMoq.Mock.ofType(QueryEditor, TypeMoq.MockBehavior.Loose, undefined, new TestThemeService(),
+			new TestStorageService(), contextkeyservice, undefined, new TestFileService(), undefined);
+		queryEditor.setup(x => x.input).returns(() => testQueryInput.object);
+		queryEditor.setup(x => x.getSelection(false)).returns(() => { return predefinedSelection; });
+
+		// Mocking runQueryStatment in unititledQueryEditorInput
+		testQueryInput.setup(x => x.runQueryStatement(TypeMoq.It.isAny())).callback(() => { calledRunQueryStatementOnInput = true; });
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/query/test/browser/queryActions.test.ts:581]: Any reason you're not just using verify to determine whether the function has been called?

File: src/sql/workbench/contrib/query/test/browser/queryActions.test.ts [hunk 4]
```diff
@@ -560,6 +560,119 @@ suite('SQL QueryAction Tests', () => {
...
+
+		// setting up queryEditor with a selection range. This case should call runQuery
+		queryEditor.setup(x => x.getSelection(false)).returns(() => { return predefinedRangeSelection; });
+		calledRunQueryOnInput = false;
+		calledRunQueryStatementOnInput = false;
+		await queryAction.runCurrent();
+
+		assert.equal(calledRunQueryStatementOnInput, false, 'runCurrent should not call runQueryStatement');
+		assert.equal(calledRunQueryOnInput, true, 'run should call runQuery');
+		testQueryInput.verify(x => x.runQueryStatement(TypeMoq.It.isAny()), TypeMoq.Times.once());
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/query/test/browser/queryActions.test.ts:673]: These tests should really be verifying that the ranges are correct so we are actually verifying that we're only running the statements that are intended

**Pull request merged**